pkg/securityscan/core/configmap.go

```diff
@@ -142,9 +143,29 @@ func getConfigMapObject(cmName, data string) *corev1.ConfigMap {
 	}
 }
 
-func getCustomBenchmarkConfigMap(benchmark *cisoperatorapiv1.ClusterScanBenchmark, configmapsClient wcorev1.ConfigMapController) (*corev1.ConfigMap, error) {
+func getCustomBenchmarkConfigMap(benchmark *cisoperatorapiv1.ClusterScanBenchmark, clusterscan *cisoperatorapiv1.ClusterScan, configmapsClient wcorev1.ConfigMapController) (*corev1.ConfigMap, error) {
 	if benchmark.Spec.CustomBenchmarkConfigMapName == "" {
 		return nil, nil
 	}
-	return configmapsClient.Get(benchmark.Spec.CustomBenchmarkConfigMapNamespace, benchmark.Spec.CustomBenchmarkConfigMapName, metav1.GetOptions{})
+	userConfigmap, err := configmapsClient.Get(benchmark.Spec.CustomBenchmarkConfigMapNamespace, benchmark.Spec.CustomBenchmarkConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if benchmark.Spec.CustomBenchmarkConfigMapNamespace == cisoperatorapiv1.ClusterScanNS {
+		return userConfigmap, nil
+	}
+	//copy the configmap to ClusterScanNS so that cis scan pod can find it for volume mount
+	//this will be cleaned up after scan job finishes
+	configmapCopy := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.SafeConcatName(cisoperatorapiv1.CustomBenchmarkConfigMap, clusterscan.Name),
+			Namespace: cisoperatorapiv1.ClusterScanNS,
+		},
+		Data: userConfigmap.Data,
+	}
+	return configmapsClient.Create(&configmapCopy)
```

Comment:
What cleans this up? Existing code I am guessing?

Reply:
Yes correct, it is the ensureCleanup in the JobHandler that cleansup after the job completes